### PR TITLE
Scroll BubbleCalendar year chart horizontally and tooltips vertically

### LINKED ISSUE #214: In BubbleCalendar, year chart increases horizontally every year
bubblecalendar ui issue

It is noticed in the bubble calendar format that year numbers are not at the same line with year chart. They are shifted one line down, because page width doesn't fit with the increasing number of years. So for each year added to the chart, page width should be increased, which is not practical. We have to find away to fix such issue as shown above.

### PR DESCRIPTION
**Pull request by MohammedElsayyed**
This pull request fixes #214 and #328 .

- Add a horizontal scrollbar to BubbleCalendar year chart to avoid exceeding window width in the future.

- Make BubbleCalendar tooltips scrollable to avoid exceeding window height when a specific day contains too many snapshots.

I merged changes done in issue #214 (PR #317) into this PR. I expect this branch 'scrollbars' will handle all scroll-related issues.

### CODE REVIEW

File: wayback-webapp/src/main/webapp/WEB-INF/query/BubbleCalendar.jsp [hunk 1]
```diff
@@ -239,7 +244,7 @@ $().ready(function(){
     
     <div class="clearfix"></div>
 
-    <div id="wbChart" onmouseout="showTrackers('none'); setActiveYear(startYear);">
+    <div id="wbChart" onmouseout="showTrackers('none'); setActiveYear(startYear);" style=" width: 963px; height: 118px;" >
```
Comment (ldko) [inline wayback-webapp/src/main/webapp/WEB-INF/query/BubbleCalendar.jsp]: superfluous space in `style=" width:`

File: wayback-webapp/src/main/webapp/WEB-INF/query/BubbleCalendar.jsp [hunk 2]
```diff
@@ -36,12 +36,11 @@ if(graphJspPrefix == null) {
...
 int imgWidth = 0;
 int imgHeight = 75;
-int yearWidth = 49;
+int yearWidth = 60;
 int monthWidth = 5;
 int startYear = Timestamp.getStartYear();
 
-for (int year = startYear; year <= Calendar.getInstance().get(Calendar.YEAR); year++)
-    imgWidth += yearWidth;
+imgWidth = yearWidth * ( Calendar.getInstance().get(Calendar.YEAR) - startYear + 1 );
```
Comment (ldko) [inline wayback-webapp/src/main/webapp/WEB-INF/query/BubbleCalendar.jsp]: Strictly a style comment, but in order to maintain consistency with the pre-existing code in this file, it makes sense to me to remove the extra whitespace after the open parentheses and before the close parentheses and in other changes made in this PR that add extra whitespace inside parentheses and curly braces.

**Comment by runderwood** [comment]
@MohammedElsayyed It doesn't appear to be part of this pull request, but it looks to me (and I'm unfamiliar w/ the code) that there's no header included in BubbleCalendar.jsp. Shouldn't there be something like this before L17?

`<jsp:include page="/WEB-INF/template/UI-header.jsp" flush="true" />`

**Comment by ldko** [comment]
@runderwood, it is problematic that we aren't including UI-header.jsp. I will open an issue to discuss this as I think handling it properly will require more than just an include of the existing header.

I also noticed that in BubbleCalendar.jsp the submit button `value` is hardcoded to `"Go Wayback!"`, but to support localization we should probably be using:
`value="<%= fmt.format("UIGlobal.urlSearchButton") %>"`
What do you think about changing that @MohammedElsayyed ?

**Comment by MohammedElsayyed** [comment]
@runderwood I agree with @ldko. Thank you for opening this issue.

@ldko of course, I will update the pull request.

**Pull request merged**